x/subspaces/types/codec.go

```diff
@@ -28,9 +28,25 @@ func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
 	cdc.RegisterConcrete(&MsgRemoveUserFromUserGroup{}, "desmos/MsgRemoveUserFromUserGroup", nil)
 
 	cdc.RegisterConcrete(&MsgSetUserPermissions{}, "desmos/MsgSetUserPermissions", nil)
+
+	cdc.RegisterConcrete(&MsgGrantUserAllowance{}, "desmos/MsgGrantUserAllowance", nil)
+	cdc.RegisterConcrete(&MsgRevokeUserAllowance{}, "desmos/MsgRevokeUserAllowance", nil)
+	cdc.RegisterConcrete(&MsgGrantGroupAllowance{}, "desmos/MsgGrantGroupAllowance", nil)
+	cdc.RegisterConcrete(&MsgRevokeGroupAllowance{}, "desmos/MsgRevokeGroupAllowance", nil)
+
+	cdc.RegisterInterface((*GrantTarget)(nil), nil)
+	cdc.RegisterConcrete(&UserTarget{}, "desmos/subspaces/UserTarget", nil)
```

Comment:
In order to avoid the name conflict with UserTarget in `x/reports`, I added the modules namespace.

Reply:
I think it's better to rename this `UserGrantee` probably, so we can avoid any name problems. We then rename the field to be `grantee` instead of `target`